### PR TITLE
Add doc for raftstore asynchronous write io

### PR DESCRIPTION
**Pull request by gengliqi**
Signed-off-by: gengliqi <gengliqiii@gmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add doc for asynchronous write io in raftstore

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by BusyJay — APPROVED** [review]

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -702,6 +702,12 @@ raftstore 相关的配置项。
 + 默认值：2
 + 最小值：大于 0
 
+### `store-io-pool-size`
+
++ 处理 raft io 任务的线程池线程数，0 表示 io 任务在 store 线程处理。
```
Comment (en-jin19) [inline tikv-configuration-file.md:707]:
```suggestion
+ 处理 Raft I/O 任务的线程池线程数量，`0` 表示 I/O 任务在处理 Raftsore 线程。
```

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -720,6 +726,18 @@ raftstore 相关的配置项。
 + 默认值：500ms
 + 最小值：1ms
 
+### `raft-write-size-limit`
+
++ 触发 raft 数据写入的阈值。`store-io-pool-size` 为 0 时不生效。
```
Comment (en-jin19) [inline tikv-configuration-file.md:731]:
```suggestion
+ 触发 Raft 数据写入的阈值。当 `store-io-pool-size` 的值为 0 时，该配置项不会生效。
```

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -720,6 +726,18 @@ raftstore 相关的配置项。
...
++ 默认值: 1MB
++ 最小值：0
+
+### `raft-msg-flush-interval`
+
++ raft 消息攒批发出的间隔时间。`store-io-pool-size` 为 0 时不生效。
++ 默认值：250us
++ 最小值：0
```
Comment (en-jin19) [inline tikv-configuration-file.md:739]:
```suggestion
### `raft-write-size-limit`

+ 触发 Raft 数据写入的阈值。当 `store-io-pool-size` 的值为 0 时，该配置项不会生效。
+ 默认值：1MB
+ 最小值：0

### `raft-msg-flush-interval`

+ Raft 消息攒批发出的间隔时间。当 `store-io-pool-size` 的值为 0 时，该配置项不会生效。
+ 默认值：250us
+ 最小值：0
```

File: tune-tikv-thread-performance.md [hunk 4]
```diff
@@ -10,11 +10,12 @@ aliases: ['/docs-cn/dev/tune-tikv-thread-performance/']
 
 ## 线程池介绍
 
-在 TiKV 中，线程池主要由 gRPC、Scheduler、UnifyReadPool、Raftstore、Apply、RocksDB 以及其它一些占用 CPU 不多的定时任务与检测组件组成，这里主要介绍几个占用 CPU 比较多且会对用户读写请求的性能产生影响的线程池。
+在 TiKV 中，线程池主要由 gRPC、Scheduler、UnifyReadPool、Raftstore、StoreWriter, Apply、RocksDB 以及其它一些占用 CPU 不多的定时任务与检测组件组成，这里主要介绍几个占用 CPU 比较多且会对用户读写请求的性能产生影响的线程池。
```
Comment (en-jin19) [inline tune-tikv-thread-performance.md:13]:
```suggestion
在 TiKV 中，线程池主要由 gRPC、Scheduler、UnifyReadPool、Raftstore、StoreWriter、Apply、RocksDB 以及其它一些占用 CPU 不多的定时任务与检测组件组成，这里主要介绍几个占用 CPU 比较多且会对用户读写请求的性能产生影响的线程池。
```

File: tune-tikv-thread-performance.md [hunk 5]
```diff
@@ -10,11 +10,12 @@ aliases: ['/docs-cn/dev/tune-tikv-thread-performance/']
...
 
 * gRPC 线程池：负责处理所有网络请求，它会把不同任务类型的请求转发给不同的线程池。
 * Scheduler 线程池：负责检测写事务冲突，把事务的两阶段提交、悲观锁上锁、事务回滚等请求转化为 key-value 对数组，然后交给 Raftstore 线程进行 Raft 日志复制。
-* Raftstore 线程池：负责处理所有的 Raft 消息以及添加新日志的提议 (Propose)、将日志写入到磁盘，当日志在多数副本中达成一致后，它就会把该日志发送给 Apply 线程。
+* Raftstore 线程池：负责处理所有的 Raft 消息以及添加新日志的提议 (Propose)、如果 `store-io-pool-size` 为 0，将日志写入到磁盘，如果不为 0，将日志发给 StoreWriter 线程、当日志在多数副本中达成一致后，它就会把该日志发送给 Apply 线程。
+* StoreWriter 线程池：该线程池负责将所有 Raft 日志写入到磁盘，再将结果发回 Raftstore 线程。
```
Comment (en-jin19) [inline tune-tikv-thread-performance.md:18]:
```suggestion
* Raftstore 线程池：负责处理所有的 Raft 消息以及添加新日志的提议 (Propose)、将日志写入到磁盘（当 `store-io-pool-size` 为 0 时）或将日志发给 StoreWriter 线程（当 `store-io-pool-size` 不是 0 时）、当日志在多数副本中达成一致后，它就会把该日志发送给 Apply 线程。
* StoreWriter 线程池：负责将所有 Raft 日志写入到磁盘后，再把结果发回到 Raftstore 线程。
```

File: tune-tikv-thread-performance.md [hunk 6]
```diff
@@ -42,9 +43,11 @@ TiKV 的读取请求分为两类：
     
     通常来说为了避免过多的线程切换，最好确保 scheduler 线程池的利用率保持在 50%～75% 之间。（如果线程池大小为 8 的话，那么 Grafana 上的 TiKV-Details.Thread CPU.scheduler worker CPU 应当在 400%～600% 之间较为合理）
 
-* Raftstore 线程池是 TiKV 最为复杂的一个线程池，默认大小 (`raftstore.store-pool-size`) 为 2，所有的写请求都会先在 Raftstore 线程 fsync 的方式写入 RocksDB。
+* Raftstore 线程池是 TiKV 最为复杂的一个线程池，默认大小 (`raftstore.store-pool-size`) 为 2。StoreWriter 线程池默认大小(`raftstore.store-io-pool-size`) 为 0。
```
Comment (en-jin19) [inline tune-tikv-thread-performance.md:46]:
```suggestion
* Raftstore 线程池是 TiKV 最为复杂的一个线程池，默认大小 (`raftstore.store-pool-size`) 为 2。StoreWriter 线程池(`raftstore.store-io-pool-size`) 的默认大小为 0。
```

File: tune-tikv-thread-performance.md [hunk 7]
```diff
@@ -42,9 +46,10 @@ TiKV 的读取请求分为两类：
...
 
-    由于存在 I/O，Raftstore 线程理论上不可能达到 100% 的 CPU。为了尽可能地减少写磁盘次数，将多个写请求攒在一起写入 RocksDB，最好控制其整体 CPU 使用在 60% 以下（按照线程数默认值 2，则 Grafana 监控上的 TiKV-Details.Thread CPU.Raft store CPU 上的数值控制在 120% 以内较为合理）。不要为了提升写性能盲目增大 Raftstore 线程池大小，这样可能会适得其反，增加了磁盘负担让性能变差。
+    * 如果 StoreWriter 线程池大小为 0，所有的写请求都会在 Raftstore 线程以 fsync 的方式写入 RocksDB。由于存在 I/O，Raftstore 线程理论上不可能达到 100% 的 CPU。为了尽可能地减少写磁盘次数，你可以使 Raftstore 线程积攒多个写请求后，再将其请求写入 RocksDB。另外，对于 Raftstore 线程的 CPU 使用率，建议最好将其整体 CPU 使用控制在 60% 以下（当把 Raftstore 线程数设置为默认值 2 时，Grafana 监控上的 TiKV-Details.Thread CPU.Raft store CPU 上的数值则适合控制在 120% 以内）。不要为了提升写性能盲目增大 Raftstore 线程池大小，这样可能会适得其反，反而增加磁盘负担，导致性能变差。
+    * 如果 StoreWriter 线程池大小不为 0，所有写请求都会在 StoreWriter 线程以 fsync 的方式写入 RocksDB。与写请求在 Raftstore 线程完成的情况相比，这样理论上能够显著地降低写延迟和读的尾延迟。然而，写入速度变得更快意味着 Raft 日志也变得更多，从而导致 Raftstore、Apply 和 gRPC 线程的 CPU 开销增多。在这种情况下，CPU 资源不足可能会抵消优化效果，反而还可能比原来的写速度更慢。因此，建议仅在整体 CPU 资源比较充裕的情况下开启此功能。由于 Raftstore 线程把绝大部分的 I/O 请求交给 StoreWriter，StoreWriter 线程的 CPU 使用率控制在 80% 以下即可。另外，由于 StoreWriter 线程池的大小会影响 Raft 日志数量，所以该值不宜过大，一般来说设为 1 或 2 即可。如果 CPU 使用率高于 80%，可以考虑再增加。此外，需要注意 Raft 日志增多对其他线程池 CPU 开销的影响，必要的时候需要相应地增加 Raftstore，Apply，gRPC 线程的大小。
```
Comment (en-jin19) [inline tune-tikv-thread-performance.md:52]:
```suggestion
    * 如果 StoreWriter 线程池大小不为 0，所有写请求都会在 StoreWriter 线程以 fsync 的方式写入 RocksDB。与写请求在 Raftstore 线程完成的情况相比，这样理论上能够显著地降低写延迟和读的尾延迟。然而，写入速度变得更快意味着 Raft 日志也变得更多，从而导致 Raftstore 线程、Apply 线程和 gRPC 线程的 CPU 开销增多。在这种情况下，CPU 资源不足可能会抵消优化效果，反而还可能比原来的写速度更慢。因此，建议仅在整体 CPU 资源比较充裕的情况下开启此功能。由于 Raftstore 线程把绝大部分的 I/O 请求交给 StoreWriter，StoreWriter 线程的 CPU 使用率控制在 80% 以下即可。另外，由于 StoreWriter 线程池的大小会影响 Raft 日志数量，所以该值不宜过大，一般来说设为 1 或 2 即可。如果 CPU 使用率高于 80%，可以考虑再增加。此外，需要注意 Raft 日志增多对其他线程池 CPU 开销的影响，必要的时候需要相应地增加 Raftstore 线程，Apply 线程和 gRPC 线程的大小。
```

File: tikv-configuration-file.md [hunk 8]
```diff
@@ -720,6 +726,18 @@ raftstore 相关的配置项。
 + 默认值：500ms
 + 最小值：1ms
 
+### `raft-write-size-limit`
+
++ 触发 Raft 数据写入的阈值。当 `store-io-pool-size` 的值为 0 时，该配置项不会生效。
```
Comment (TomShawn) [inline tikv-configuration-file.md:731]: 从配置项单词上猜的，求看~
```suggestion
+ 触发写入 Raft 数据的数据大小限制。当数据大小超过该配置项值，数据不会被写入。当 `store-io-pool-size` 的值为 `0` 时，该配置项不生效。
```

File: tikv-configuration-file.md [hunk 9]
```diff
@@ -720,6 +726,18 @@ raftstore 相关的配置项。
...
++ 默认值：1MB
++ 最小值：0
+
+### `raft-msg-flush-interval`
+
++ Raft 消息攒批发出的间隔时间。当 `store-io-pool-size` 的值为 0 时，该配置项不会生效。
```
Comment (TomShawn) [inline tikv-configuration-file.md:737]:
```suggestion
+ Raft 消息攒批发出的间隔时间。每隔该配置项指定的间隔，Raft 消息会批量发出。当 `store-io-pool-size` 的值为 0 时，该配置项不会生效。
```

File: tikv-configuration-file.md (hunk 8 above)
Comment (gengliqi) [reply tikv-configuration-file.md:731]: 啊其实意思是超过该配置项，数据会被写入，配置命名貌似不太好。。。

File: tune-tikv-thread-performance.md [hunk 10]
```diff
@@ -42,9 +46,10 @@ TiKV 的读取请求分为两类：
     
     通常来说为了避免过多的线程切换，最好确保 scheduler 线程池的利用率保持在 50%～75% 之间。（如果线程池大小为 8 的话，那么 Grafana 上的 TiKV-Details.Thread CPU.scheduler worker CPU 应当在 400%～600% 之间较为合理）
 
-* Raftstore 线程池是 TiKV 最为复杂的一个线程池，默认大小 (`raftstore.store-pool-size`) 为 2，所有的写请求都会先在 Raftstore 线程 fsync 的方式写入 RocksDB。
+* Raftstore 线程池是 TiKV 最为复杂的一个线程池，默认大小 (`raftstore.store-pool-size`) 为 2。StoreWriter 线程池 (`raftstore.store-io-pool-size`) 的默认大小为 0。
```
Comment (TomShawn) [inline tune-tikv-thread-performance.md:49]:
```suggestion
* Raftstore 线程池是 TiKV 中最复杂的一个线程池，默认大小 (由 `raftstore.store-pool-size` 控制) 为 2。StoreWriter 线程池 (由 `raftstore.store-io-pool-size` 控制) 的默认大小为 0。
```

File: tune-tikv-thread-performance.md [hunk 11]
```diff
@@ -42,9 +46,10 @@ TiKV 的读取请求分为两类：
...
 
-    由于存在 I/O，Raftstore 线程理论上不可能达到 100% 的 CPU。为了尽可能地减少写磁盘次数，将多个写请求攒在一起写入 RocksDB，最好控制其整体 CPU 使用在 60% 以下（按照线程数默认值 2，则 Grafana 监控上的 TiKV-Details.Thread CPU.Raft store CPU 上的数值控制在 120% 以内较为合理）。不要为了提升写性能盲目增大 Raftstore 线程池大小，这样可能会适得其反，增加了磁盘负担让性能变差。
+    * 如果 StoreWriter 线程池大小为 0，所有的写请求都会在 Raftstore 线程以 fsync 的方式写入 RocksDB。由于存在 I/O，Raftstore 线程理论上不可能达到 100% 的 CPU。为了尽可能地减少写磁盘次数，你可以使 Raftstore 线程积攒多个写请求后，再将其请求写入 RocksDB。另外，对于 Raftstore 线程的 CPU 使用率，建议最好将其整体 CPU 使用控制在 60% 以下（当把 Raftstore 线程数设置为默认值 2 时，Grafana 监控上的 TiKV-Details.Thread CPU.Raft store CPU 上的数值则适合控制在 120% 以内）。不要为了提升写性能盲目增大 Raftstore 线程池大小，这样可能会适得其反，反而增加磁盘负担，导致性能变差。
```
Comment (TomShawn) [inline tune-tikv-thread-performance.md:51]:
```suggestion
    * 当 StoreWriter 线程池大小为 0，所有的写请求都会由 Raftstore 线程以 fsync 的方式写入 RocksDB。因此建议：
        * 使 Raftstore 线程积攒多个写请求后，再将积攒的请求写入 RocksDB。这样能尽可能地减少写入磁盘的次数。由于存在 I/O 请求，理论上 Raftstore 线程的 CPU 使用率总是低于 100%。
        * 将 Raftstore 线程的整体 CPU 使用率控制在 60% 以下。当把 Raftstore 线程数设为默认值 2 时，Grafana 监控上的 **TiKV-Details**、**Thread CPU**、**Raft store CPU** 面版上的数值则适合控制在 120% 以内。
        * 不建议为了提升写性能而盲目增大 Raftstore 线程池大小，这样可能会适得其反，增加磁盘负担，导致性能变差。
```

File: tune-tikv-thread-performance.md [hunk 12]
```diff
@@ -42,9 +46,10 @@ TiKV 的读取请求分为两类：
...
+    * 如果 StoreWriter 线程池大小不为 0，所有写请求都会在 StoreWriter 线程以 fsync 的方式写入 RocksDB。与写请求在 Raftstore 线程完成的情况相比，这样理论上能够显著地降低写延迟和读的尾延迟。然而，写入速度变得更快意味着 Raft 日志也变得更多，从而导致 Raftstore 线程、Apply 线程和 gRPC 线程的 CPU 开销增多。在这种情况下，CPU 资源不足可能会抵消优化效果，反而还可能比原来的写速度更慢。因此，建议仅在整体 CPU 资源比较充裕的情况下开启此功能。由于 Raftstore 线程把绝大部分的 I/O 请求交给 StoreWriter，StoreWriter 线程的 CPU 使用率控制在 80% 以下即可。另外，由于 StoreWriter 线程池的大小会影响 Raft 日志数量，所以该值不宜过大，一般来说设为 1 或 2 即可。如果 CPU 使用率高于 80%，可以考虑再增加。此外，需要注意 Raft 日志增多对其他线程池 CPU 开销的影响，必要的时候需要相应地增加 Raftstore 线程，Apply 线程和 gRPC 线程的大小。
```
Comment (TomShawn) [inline tune-tikv-thread-performance.md:52]:
```suggestion
    * 当 StoreWriter 线程池大小不为 0 时，所有写请求都由 StoreWriter 线程以 fsync 的方式写入 RocksDB。此时建议：
        * 仅在整体 CPU 资源比较充裕的情况下启用 StoreWriter 线程池，并将 StoreWriter 线程的 CPU 使用率控制在 80% 以下。
            与写请求在 Raftstore 线程完成的情况相比，理论上 StoreWriter 线程处理写请求能够显著地降低写延迟和读的尾延迟。然而，写入速度变得更快意味着 Raft 日志也变得更多，从而导致 Raftstore 线程、Apply 线程和 gRPC 线程的 CPU 开销增多。在这种情况下，CPU 资源不足可能会抵消优化效果，反而还可能比原来的写速度更慢，因此若是 CPU 资源不充裕则不建议使用 StoreWriter 线程。由于 Raftstore 线程把绝大部分的 I/O 请求交给 StoreWriter，因此可将 StoreWriter 线程的 CPU 使用率控制在 80% 以下。
        * 将 StoreWriter 线程池的大小设为 1 或 2 即可。这是因为 StoreWriter 线程池的大小会影响 Raft 日志数量，所以该值不宜过大。如果 CPU 使用率高于 80%，可以考虑再增加其大小。
        * 注意 Raft 日志增多对其他线程池 CPU 开销的影响，必要的时候需要相应地增加 Raftstore 线程、Apply 线程和 gRPC 线程的数量。
```

File: tune-tikv-thread-performance.md (hunk 11 above)
Comment (gengliqi) [reply tune-tikv-thread-performance.md]: * 使 Raftstore 线程积攒多个写请求后，再将积攒的请求写入 RocksDB。这样能尽可能地减少写入磁盘的次数。这句话我删了，因为这个是内部机制，用户控制不了。我把 `由于存在 I/O 请求，理论上 Raftstore 线程的 CPU 使用率总是低于 100%` 这句话移到第二句的下面了

File: tune-tikv-thread-performance.md (hunk 12 above)
Comment (gengliqi) [reply tune-tikv-thread-performance.md]: `由于 Raftstore 线程把绝大部分的 I/O 请求交给 StoreWriter，因此可将 StoreWriter 线程的 CPU 使用率控制在 80% 以下。` 上次修改的时候没注意看。。这里意思写错了，这里的想说的是 Raftstore 线程的 CPU 使用率在 80% 以下。

**Review by TomShawn — APPROVED** [review]
LGTM

File: tikv-configuration-file.md [hunk 13]
```diff
@@ -698,13 +698,19 @@ raftstore 相关的配置项。
 
 ### `store-pool-size`
 
-+ 处理 raft 的线程池线程数。
++ 处理 Raft 的线程池中线程的数量。
 + 默认值：2
 + 最小值：大于 0
 
+### `store-io-pool-size`
```
Comment (en-jin19) [inline tikv-configuration-file.md:705]:
```suggestion
### `store-io-pool-size`<span class="version-mark">从 v5.3 版本开始引入</span>
```

File: tikv-configuration-file.md [hunk 14]
```diff
@@ -698,13 +698,19 @@ raftstore 相关的配置项。
 
 ### `store-pool-size`
 
-+ 处理 raft 的线程池线程数。
++ 处理 Raft 的线程池中线程的数量。
```
Comment (en-jin19) [inline tikv-configuration-file.md:701]:
```suggestion
+ 处理 Raft 的线程池中线程的数量。当 [`store-io-pool-size`](/#store-io-pool-size) 大于默认值 `0` 时，可能需要对该配置项也做出相应地调整。具体内容，请参见 [TiKV 线程池调优](/tune-tikv-thread-performance.md#tikv-线程池调优)。
```

File: tikv-configuration-file.md [hunk 15]
```diff
@@ -720,6 +726,18 @@ raftstore 相关的配置项。
...
+
++ 触发 Raft 数据写入的阈值。当数据大小超过该配置项值，数据会被写入磁盘。当 `store-io-pool-size` 的值为 `0` 时，该配置项不生效。
++ 默认值：1MB
++ 最小值：0
+
+### `raft-msg-flush-interval`
+
++ Raft 消息攒批发出的间隔时间。每隔该配置项指定的间隔，Raft 消息会攒批发出。当 `store-io-pool-size` 的值为 `0` 时，该配置项不会生效。
++ 默认值：250us
++ 最小值：0
```
Comment (en-jin19) [inline tikv-configuration-file.md:739]:
```suggestion
### `raft-write-size-limit`<span class="version-mark">从 v5.3 版本开始引入</span>

+ 触发 Raft 数据写入的阈值。当数据大小超过该配置项值，数据会被写入磁盘。当 `store-io-pool-size` 的值为 `0` 时，该配置项不生效。
+ 默认值：1MB
+ 最小值：0

### `raft-msg-flush-interval`<span class="version-mark">从 v5.3 版本开始引入</span>

+ Raft 消息攒批发出的间隔时间。每隔该配置项指定的间隔，Raft 消息会攒批发出。当 `store-io-pool-size` 的值为 `0` 时，该配置项不会生效。
+ 默认值：250us
+ 最小值：0
```

File: tikv-configuration-file.md (hunk 14 above)
Comment (TomShawn) [reply tikv-configuration-file.md:701]:
```suggestion
+ 表示 Raftstore 中 执行 CPU 任务的线程数量。
```

File: tikv-configuration-file.md [hunk 16]
```diff
@@ -698,13 +698,19 @@ raftstore 相关的配置项。
...
 ### `store-pool-size`
 
-+ 处理 raft 的线程池线程数。
++ 处理 Raft 的线程池中线程的数量。
 + 默认值：2
 + 最小值：大于 0
 
+### `store-io-pool-size`
+
++ 处理 Raft I/O 任务的线程池中线程的数量，`0` 表示 I/O 任务由 Raftstore 线程处理。
```
Comment (TomShawn) [inline tikv-configuration-file.md:707]:
```suggestion
+ 处理 Raft I/O 任务的线程池中线程的数量，即 StoreWriter 线程池的大小。`0` 表示 I/O 任务由 Raftstore 线程处理。
```

File: tikv-configuration-file.md (hunk 14 above)
Comment (en-jin19) [reply tikv-configuration-file.md:701]:
```suggestion
	+ 处理 Raft 的线程池中线程的数量，即 Raftstore 线程池的大小。
	+ 当 [`store-io-pool-size`](#store-io-pool-size) 配置项的值为 `0` 时，由 Raftstore 线程执行 I/O 任务，将日志写入到磁盘。当 `store-io-pool-size`](#store-io-pool-size) 配置项的值大于  `0` 时，StoreWriter 线程将进行日志日志写入，此时，可能需要调整 `store-pool-size` 配置项的值。具体内容，请参考 [TiKV 线程池调优](/tune-tikv-thread-performance.md#tikv-线程池调优)。
```

File: tikv-configuration-file.md [hunk 17]
```diff
@@ -686,7 +686,7 @@ raftstore 相关的配置项。
 
 ### `apply-pool-size`
 
-+ 处理数据落盘的线程池线程数。
++ 处理数据落盘的线程池中线程的数量。
```
Comment (en-jin19) [inline tikv-configuration-file.md:689]:
```suggestion
+ 处理数据落盘的线程池中线程的数量。调整该线程池的大小时，请参考 [TiKV 线程池调优](/tune-tikv-thread-performance.md#tikv-线程池调优)。
```

File: tikv-configuration-file.md [hunk 18]
```diff
@@ -99,7 +99,7 @@ TiKV 配置文件比命令行参数支持更多的选项。你可以在 [etc/con
 
 ### `grpc-raft-conn-num`
 
-+ tikv 节点之间用于 raft 通讯的链接最大数量。
++ tikv 节点之间用于 Raft 通讯的链接最大数量。
```
Comment (TomShawn) [inline tikv-configuration-file.md:102]:
```suggestion
+ TiKV 节点之间用于 Raft 通讯的链接最大数量。
```

File: tikv-configuration-file.md [hunk 19]
```diff
@@ -182,7 +182,7 @@ TiKV 配置文件比命令行参数支持更多的选项。你可以在 [etc/con
 
 ### `max-thread-count`
 
-+ 统一处理读请求的线程池最多的线程数量。
++ 统一处理读请求的线程池最多的线程数量。调整 UnifyReadPool 线程池的大小时，请参考 [TiKV 线程池调优](/tune-tikv-thread-performance.md#tikv-线程池调优)。
```
Comment (TomShawn) [inline tikv-configuration-file.md:185]:
```suggestion
+ 统一处理读请求的线程池最多的线程数量，即 UnifyReadPool 线程池的大小。调整该线程池的大小时，请参考 [TiKV 线程池调优](/tune-tikv-thread-performance.md#tikv-线程池调优)。
```

File: tikv-configuration-file.md [hunk 20]
```diff
@@ -686,7 +686,7 @@ raftstore 相关的配置项。
 
 ### `apply-pool-size`
 
-+ 处理数据落盘的线程池线程数。
++ 处理数据落盘的线程池中线程的数量。调整 Apply 线程池的大小时，请参考 [TiKV 线程池调优](/tune-tikv-thread-performance.md#tikv-线程池调优)。
```
Comment (TomShawn) [inline tikv-configuration-file.md:689]:
```suggestion
+ 处理数据落盘的线程池中线程的数量。调整该线程池的大小时，请参考 [TiKV 线程池调优](/tune-tikv-thread-performance.md#tikv-线程池调优)。
```

**Review by TomShawn — APPROVED** [review]
Rest LGTM

File: tune-tikv-thread-performance.md [hunk 21]
```diff
@@ -10,11 +10,15 @@ aliases: ['/docs-cn/dev/tune-tikv-thread-performance/']
...
 
-在 TiKV 中，线程池主要由 gRPC、Scheduler、UnifyReadPool、Raftstore、Apply、RocksDB 以及其它一些占用 CPU 不多的定时任务与检测组件组成，这里主要介绍几个占用 CPU 比较多且会对用户读写请求的性能产生影响的线程池。
+在 TiKV 中，线程池主要由 gRPC、Scheduler、UnifyReadPool、Raftstore、StoreWriter、Apply、RocksDB 以及其它一些占用 CPU 不多的定时任务与检测组件组成，这里主要介绍几个占用 CPU 比较多且会对用户读写请求的性能产生影响的线程池。
 
 * gRPC 线程池：负责处理所有网络请求，它会把不同任务类型的请求转发给不同的线程池。
 * Scheduler 线程池：负责检测写事务冲突，把事务的两阶段提交、悲观锁上锁、事务回滚等请求转化为 key-value 对数组，然后交给 Raftstore 线程进行 Raft 日志复制。
-* Raftstore 线程池：负责处理所有的 Raft 消息以及添加新日志的提议 (Propose)、将日志写入到磁盘，当日志在多数副本中达成一致后，它就会把该日志发送给 Apply 线程。
+* Raftstore 线程池：
+    * 处理所有的 Raft 消息以及添加新日志的提议 (Propose)。
+    * 处理 Raft 日志。如果 [`store-io-pool-size`](/tikv-configuration-file.md#store-io-pool-size) 配置项的值为 `0`，Raftstore 线程将日志写入到磁盘；如果该值不为 `0`，Raftstore 线程将日志发送给 StoreWriter 线程处理。
```
Comment (TomShawn) [inline tune-tikv-thread-performance.md:19]:
```suggestion
    * 处理 Raft 日志。如果 [`store-io-pool-size`](/tikv-configuration-file.md#store-io-pool-size-从-v530-版本开始引入) 配置项的值为 `0`，Raftstore 线程将日志写入到磁盘；如果该值不为 `0`，Raftstore 线程将日志发送给 StoreWriter 线程处理。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Comment by TomShawn** [comment]
@en-jin19 Please fix the CI failure. Thanks~

File: tikv-configuration-file.md [hunk 22]
```diff
@@ -698,13 +698,19 @@ raftstore 相关的配置项。
 
 ### `store-pool-size`
 
-+ 处理 raft 的线程池线程数。
++ 表示处理 Raft 的线程池中线程的数量，即 Raftstore 线程池的大小。调整该线程池的大小时，请参考 [TiKV 线程池调优](/tune-tikv-thread-performance.md#tikv-线程池调优)。
 + 默认值：2
 + 最小值：大于 0
 
+### `store-io-pool-size`<span class="version-mark">从 v5.3.0 版本开始引入</span>
```
Comment (en-jin19) [inline tikv-configuration-file.md:705]:
```suggestion
### `store-io-pool-size`<span class="version-mark"> 从 v5.3.0 版本开始引入</span>
```

File: tikv-configuration-file.md [hunk 23]
```diff
@@ -720,6 +726,18 @@ raftstore 相关的配置项。
 + 默认值：500ms
 + 最小值：1ms
 
+### `raft-write-size-limit`<span class="version-mark">从 v5.3.0 版本开始引入</span>
```
Comment (en-jin19) [inline tikv-configuration-file.md:729]:
```suggestion
### `raft-write-size-limit`<span class="version-mark"> 从 v5.3.0 版本开始引入</span>
```

File: tikv-configuration-file.md [hunk 24]
```diff
@@ -720,6 +726,18 @@ raftstore 相关的配置项。
...
+
++ 触发 Raft 数据写入的阈值。当数据大小超过该配置项值，数据会被写入磁盘。当 `store-io-pool-size` 的值为 `0` 时，该配置项不生效。
++ 默认值：1MB
++ 最小值：0
+
+### `raft-msg-flush-interval`<span class="version-mark">从 v5.3.0 版本开始引入</span>
```
Comment (en-jin19) [inline tikv-configuration-file.md:735]:
```suggestion
### `raft-msg-flush-interval`<span class="version-mark"> 从 v5.3.0 版本开始引入</span>
```

**Comment by en-jin19** [comment]
> @en-jin19 Please fix the CI failure. Thanks~

have fixed, PTAL, thanks~

File: tikv-configuration-file.md [hunk 25]
```diff
@@ -698,13 +698,19 @@ raftstore 相关的配置项。
 
 ### `store-pool-size`
 
-+ 处理 raft 的线程池线程数。
++ 表示处理 Raft 的线程池中线程的数量，即 Raftstore 线程池的大小。调整该线程池的大小时，请参考 [TiKV 线程池调优](/tune-tikv-thread-performance.md#tikv-线程池调优)。
 + 默认值：2
 + 最小值：大于 0
 
+### `store-io-pool-size`<span class="version-mark"> 从 v5.3.0 版本开始引入</span>
```
Comment (en-jin19) [inline tikv-configuration-file.md:705]:
```suggestion
### `store-io-pool-size` <span class="version-mark">从 v5.3.0 版本开始引入</span>
```

File: tikv-configuration-file.md [hunk 26]
```diff
@@ -720,6 +726,18 @@ raftstore 相关的配置项。
 + 默认值：500ms
 + 最小值：1ms
 
+### `raft-write-size-limit`<span class="version-mark"> 从 v5.3.0 版本开始引入</span>
```
Comment (en-jin19) [inline tikv-configuration-file.md:729]:
```suggestion
### `raft-write-size-limit` <span class="version-mark">从 v5.3.0 版本开始引入</span>
```

File: tikv-configuration-file.md [hunk 27]
```diff
@@ -720,6 +726,18 @@ raftstore 相关的配置项。
...
+
++ 触发 Raft 数据写入的阈值。当数据大小超过该配置项值，数据会被写入磁盘。当 `store-io-pool-size` 的值为 `0` 时，该配置项不生效。
++ 默认值：1MB
++ 最小值：0
+
+### `raft-msg-flush-interval`<span class="version-mark"> 从 v5.3.0 版本开始引入</span>
```
Comment (en-jin19) [inline tikv-configuration-file.md:735]:
```suggestion
### `raft-msg-flush-interval` <span class="version-mark">从 v5.3.0 版本开始引入</span>
```

**Pull request merged**